packages/ui/src/composites/wui-list-account/index.ts

```diff
@@ -0,0 +1,125 @@
+import { html, LitElement } from 'lit'
+import { property } from 'lit/decorators.js'
+import '../../components/wui-text/index.js'
+import '../../components/wui-image/index.js'
+import '../../layout/wui-flex/index.js'
+import { elementStyles, resetStyles } from '../../utils/ThemeUtil.js'
+import { customElement } from '../../utils/WebComponentsUtil.js'
+import styles from './styles.js'
+import { UiHelperUtil } from '../../utils/UiHelperUtil.js'
+import { W3mFrameRpcConstants } from '@web3modal/wallet'
+import {
+  AccountController,
+  BlockchainApiController,
+  NetworkController,
+  StorageUtil
+} from '@web3modal/core'
+
+@customElement('wui-list-account')
+export class WuiListAccount extends LitElement {
+  public static override styles = [resetStyles, elementStyles, styles]
+
+  // -- State & Properties -------------------------------- //
+  @property() public accountAddress = ''
+
+  @property() public accountType = ''
+
+  private connectedConnector = StorageUtil.getConnectedConnector()
+  private labels = AccountController.state.addressLabels
+  private balance = 0
+  private fetchingBalance = true
+  private shouldShowIcon = false
+  private caipNetwork = NetworkController.state.caipNetwork
+  @property({ type: Boolean }) public selected = false
+
+  @property({ type: Function }) public onSelect?: (
+    { address, type }: { address: string; type: string },
+    selected: boolean
+  ) => void
+
+  handleClick = (event: Event) => {
+    this.onSelect?.(
+      { address: this.accountAddress, type: this.accountType },
+      // @ts-expect-error - checked is available on the event
+      event?.target?.checked
+    )
+  }
+
+  public override connectedCallback() {
+    super.connectedCallback()
+    BlockchainApiController.getBalance(this.accountAddress, this.caipNetwork?.id).then(response => {
+      let total = this.balance
+      if (response.balances.length > 0) {
+        total = response.balances.reduce((acc, balance) => acc + (balance?.value || 0), 0)
+      }
+      this.balance = total
+      this.fetchingBalance = false
+      this.requestUpdate()
+    })
+  }
+  // -- Render -------------------------------------------- //
+  public override render() {
+    let label = this.labels?.get(this.accountAddress)
+
+    // If there is no provided label, set one depending on the account type
+    if (!label && this.connectedConnector === 'AUTH') {
+      label = `${this.accountType === 'eoa' ? 'Email' : 'Smart'} Account`
+    } else if (
+      (!label && this.connectedConnector === 'INJECTED') ||
+      this.connectedConnector === 'ANNOUNCED'
+    ) {
+      label = `Injected Account`
+    } else if (!label) {
+      label = 'EOA'
+    }
```

Comment:
Can we separate this logic into another private function like `getLabel()` and just call it in the render? Main render function should be as simple as possible to make the code more modular and maintainable.